mne/source_space.py

```diff
@@ -2275,6 +2275,112 @@ def get_volume_labels_from_aseg(mgz_fname):
+    """Helper to get a nearest-neigbor vertex match"""
+    # adapted from mne_make_source_space.c, knowing accurate=False (i.e.
+    # nearest-neighbor mode should be used)
+    morph_inuse = np.zeros(morph['np'], int)
+    best = np.zeros(len(src['vertno']), int)
+    ones = _compute_nearest(rrs[1], rrs[0][src['vertno']])
+    for v, one in zip(src['vertno'], ones):
+        # if it were actually a proper morph map, we would do this, but since
+        # we know it's nearest neighbor list, we don't need to:
+        # this_mm = mm[v]
+        # one = this_mm.indices[this_mm.data.argmax()]
+        if morph_inuse[one]:
+            # Try the nearest neighbors
+            neigh = _get_surf_neighbors(morph, one)  # on demand calc
+            was = one
+            one = neigh[np.where(~morph_inuse[neigh])[0]]
+            if len(one) == 0:
+                raise RuntimeError('vertex %d would be used multiple times.'
+                                   % one)
+            one = one[0]
+            logger.info('Source space vertex moved from %d to %d because of '
+                        'double occupation.' % (was, one))
+        best[v] = one
+        morph_inuse[one] = True
+    return best
+
+
+@verbose
+def morph_source_spaces(src_from, subject_to, surf='white', subjects_dir=None,
+                        verbose=None):
+    """Morph an existing source space to a different subject
+
+    .. note:: This can be used in place of morphing source estimates for
+              multiple subjects, but there may be consequences in terms
+              of dipole topology.
+
+    Parameters
+    ----------
+    src_from : instance of SourceSpaces
+        Surface source spaces to morph.
+    subject_to : str
+        The destination subject.
+    surf : str
+        The brain surface to use for the new source space.
+    subjects_dir : string, or None
+        Path to SUBJECTS_DIR if it is not set in the environment.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    src : instance of SourceSpaces
+        The morphed source spaces.
+    """
+    # adapted from mne_make_source_space.c
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    src_out = list()
+    for fro in src_from:
+        hemi, idx = _get_hemi(fro)
+        to = op.join(subjects_dir, subject_to, 'surf', '%s.%s' % (hemi, surf,))
+        logger.info('Reading destination surface %s' % (to,))
+        to = _read_surface_geom(to, patch_stats=False, verbose=False)
+        to['rr'] /= 1000.  # convert to meters
+        _complete_surface_info(to, verbose=False)  # get normals
+        subject_from = fro['subject_his_id']
+        # Now we morph the vertices to the destination
+        logger.info('Reading %s<->%s morph maps'
+                    % (subject_from, subject_to))
+        # mm = read_morph_map(subject_from, subject_to, subjects_dir)
```

Comment:
I left this in so people looking at the C code wouldn't be mystified by what's going on. So instead of deleting this, I made comments around it clearer why it's been left in, and why it isn't used.